cunumeric/array.py

```diff
@@ -1559,24 +1559,21 @@ def argmax(self, axis=None, out=None):
         Multiple GPUs, Multiple CPUs
 
         """
-        if self.size == 1:
-            return 0
-        if axis is None:
-            axis = self.ndim - 1
-        elif type(axis) != int:
-            raise TypeError("'axis' argument for argmax must be an 'int'")
-        elif axis < 0 or axis >= self.ndim:
-            raise TypeError("invalid 'axis' argument for argmax " + str(axis))
+        if out is not None and out.dtype != np.int64:
+            raise ValueError("output array must have int64 dtype")
+        if axis is not None and not isinstance(axis, int):
+            raise ValueError("axis must be an integer")
```

Comment:
Do you need to check these here (and in argmin)? Aren't these taken care of by `_perform_unary_reduction`?

Reply:
`argmin` and `argmax` have a stronger requirement; they don't allow the `axis` to be a tuple. That's why this extra check is in place.